linalg/operator.cpp

```diff
@@ -628,7 +628,10 @@ double PowerMethod::EstimateLargestEigenvalue(Operator& opr, Vector& v0,
                                               int numSteps, double tolerance, int seed)
 {
    v1.SetSize(v0.Size());
-   v0.Randomize(seed);
+   if (seed != -1)
+   {
+      v0.Randomize(seed);
```

Comment:
Note that skipping this could be dangerous from math perspective. The goal of the randomization is to introduce a variety in `v0` because otherwise we may be stuck at an eigenvector.

Reply:
We currently generate a random vector once then set `v0` to it each time before calling this, is that problematic?

Reply:
We need a variety of values in `v0`, so if it is already randomized, skipping is fine

Reply:
Of course if we provide the same `v0` for the same `opr` the results will be the same.

But if `opr` changes, `v0` doesn't have to (provided again that it is really randomized)

Reply:
`opr` is new each time. I'm going to add a note in the header